lumen_runtime/src/otp/erlang/base.rs

```diff
@@ -41,10 +45,20 @@ impl TryFrom<TypedTerm> for Base {
                 {
                     Ok(Base(small_integer_isize as u8))
                 } else {
-                    Err(badarg!().into())
+                    Err(TryFromTypedTermError::OutOfRange)
                 }
             }
-            _ => Err(badarg!().into()),
+            _ => Err(TryFromTypedTermError::Type),
         }
     }
 }
+
+pub enum TryFromTermError {
```

Comment:
I thought we already had a type for this? If not, we should define it in `liblumen_alloc` and just consume it here